service/storage_service.cc

```diff
@@ -6250,7 +6250,15 @@ storage_service::get_natural_endpoints(const sstring& keyspace,
     auto schema = _db.local().find_schema(keyspace, cf);
     partition_key pk = partition_key::from_nodetool_style_string(schema, key);
     dht::token token = schema->get_partitioner().get_token(*schema, pk.view());
-    return _db.local().find_keyspace(keyspace).get_vnode_effective_replication_map()->get_natural_endpoints(token);
+    const auto& ks = _db.local().find_keyspace(keyspace);
+    if (ks.uses_tablets()) {
+        const auto& topology = get_token_metadata().get_topology();
+        const auto& tm = get_token_metadata().tablets().get_tablet_map(schema->id());
+        return boost::copy_range<inet_address_vector_replica_set>(tm.get_tablet_info(token).replicas | boost::adaptors::transformed([&topology] (const locator::tablet_replica& r) {
+            return topology.get_node(r.host).endpoint();
+        }));
```

Comment:
Better to use get_natural_endpoints() from table's erm. For symmetry with vnode case, and brevity.

Reply:
Fixed in v2